subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -140,9 +140,22 @@ class ArtifactTransformValuesInjectionIntegrationTest extends AbstractDependency
         settingsFile << """
             include 'a', 'b', 'c'
         """
+        buildFile << """
+            interface NestedType {
+                @InputFile
+                RegularFileProperty getInputFile()
+                @OutputDirectory
+                DirectoryProperty getOutputDirectory()
+            }
+                        
+            allprojects {
+                ext.nested = objects.newInstance(NestedType)
+            }
+        """
         setupBuildWithColorTransform {
             params("""
                 extension = 'green'
+                nested.set(project.ext.nested)
```

Comment:
Hm, is this how nested properties are to be used? Why don't we do this automatically?

Reply:
What shall we do automatically? Create the nested bean automatically from the interface type? That is mostly not what you want - given that the nested bean probably comes from somewhere else.

Reply:
Seems like in at least some cases that's what you'd want. But if we don't want to support interfaces here, why don't we use a concrete class instead?

Reply:
It needs to be a managed type, so it needs to be generated by Gradle from an interface. If the type is not managed, we cannot isolate it.

The `Property` is generated automatically, it is not present as usual. That is also the default behavior for `Property`s, and I would not change that. Could you give me a code example what you would expect to work?